Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -135,11 +135,6 @@ public ActionsTreeView(VisualElement root, StateContainer stateContainer)
                     var item = m_ActionsTreeView.GetItemDataForIndex<ActionOrBindingData>(m_ActionsTreeView.selectedIndex);
                     Dispatch(item.isAction ? Commands.SelectAction(item.name) : Commands.SelectBinding(item.bindingIndex));
                 }
```

Comment:
I guess this is safe, the problem here is that there seem to be two related but separate concepts of selection, one in the UI model part of the UITK controls and one in the custom model part of this repo.

Reply:
My only worry is that when the number of actions goes down to zero, as in deleting the last action form a map, the UI might be without selection but the State model might have a selection so they are out of sync. I suspect the loop is due to the below commands when being handled by the UI code do not skip further processing if the UI model and this custom selection model are actually equal. E.g. that else should maybe be an else if checking if the selections are not in sync.